src/maps/components/AddressPicker.vue

```diff
@@ -1,66 +1,64 @@
 <template>
   <div>
-    <QInput
-      :value="value.address"
+    <QSelect
+      use-input
+      fill-input
       clearable
-      :placeholder="$t('BUTTON.SEARCH')"
+      hide-dropdown-icon
       :label="label"
       :error="error"
       :error-message="errorMessage"
+      :options="options"
+      :placeholder="$t('BUTTON.SEARCH')"
+      :value="value.address"
+      :input-debounce="1000"
       @input="input"
-      @keyup.esc="$refs.menu.hide()"
-      @keyup.enter.prevent.stop="search"
+      @filter="search"
     >
-      <template v-slot:append>
-        <QBtn
-          v-if="value.address"
-          icon="fas fa-search"
-          :title="$t('BUTTON.SEARCH')"
-          flat
-          round
-          size="sm"
-          @click="search"
-        />
-      </template>
-      <template v-slot:before>
+      <template #before>
         <QIcon :name="icon" />
       </template>
-      <QMenu
-        ref="menu"
-        fit
-        no-parent-event
-        square
-      >
-        <QList>
-          <QItem
-            v-for="(result, idx) in options"
-            :key="idx"
-            v-close-popup
-            clickable
-            @click="select(result)"
-          >
-            <QItemSection>
-              <QItemLabel>
-                {{ result.label }}
-              </QItemLabel>
-            </QItemSection>
-          </QItem>
-          <QItem
-            v-if="options.length < 1"
-          >
-            <QItemSection>
-              <QItemLabel>
-                {{ $t('GLOBAL.SEARCH_NOT_FOUND') }}
-              </QItemLabel>
-            </QItemSection>
-          </QItem>
-        </QList>
-      </QMenu>
-    </QInput>
+      <template #selected-item />
+      <template #option="{ index, itemProps, itemEvents, opt: { label: itemLabel, useSearchTerm } }">
+        <QItem
+          :key="index"
+          v-bind="itemProps"
+          v-on="itemEvents"
+        >
+          <QItemSection>
+            <QItemLabel v-if="useSearchTerm">
+              {{ $t('ADDRESS_PICKER.SET_AS') }}
+              <strong>{{ itemLabel }}</strong>
+              <span v-if="hasLocation">({{ $t('ADDRESS_PICKER.KEEP_EXISTING_LOCATION') }})</span>
+              <span v-else>({{ $t('ADDRESS_PICKER.CHOOSE_LOCATION') }})</span>
+            </QItemLabel>
+            <QItemLabel v-else>
+              {{ itemLabel }}
+            </QItemLabel>
+          </QItemSection>
+        </QItem>
+        <QSeparator v-if="useSearchTerm" />
```

Comment:
Maybe move everything below inside a `<template v-if="useSearchTearm">` to make the meaning more obvious/less repetitive.